### PR TITLE
Add placeholder to numeric_input

### PR DESCRIPTION
**Pull request by osenan**
Short description (with a reference to an issue).

This PR closes the following issue: #428 

It includes a change in the tag creation of the `numeric_input`
Additionally, the checks that the placeholder type is character.
It also has the updated unit tests to check as well the placeholder argument.

This app creates a placeholder as `value` in the `numeric_input` is NULL
```
library(shiny)
library(shiny.semantic)
ui <- semanticPage(
  numeric_input("ex", "Select number", value = NULL, placeholder = "type value")
)
server <- function(input, output, session) {}
shinyApp(ui, server)
```
image

**DoD**

- [ ] Major project work has a corresponding task. If there’s no task for what you are doing, create it. Each task needs to be well defined and described.

- [ ] Change has been tested (manually or with automated tests), everything runs correctly and works as expected. No existing functionality is broken.

- [ ] No new error or warning messages are introduced.

- [ ] All interaction with a semantic functions, examples and docs are written from the perspective of the person using or receiving it. They are understandable and helpful to this person.

- [ ] If the change affects code or repo sctructure, README, documentation and code comments should be updated. 

- [ ] All code has been peer-reviewed before merging into any main branch.

- [ ] All changes have been merged into the main branch we use for development (develop).

- [ ] Continuous integration checks (linter, unit tests) are configured and passed.

- [ ] Unit tests added for all new or changed logic.

- [ ] All task requirements satisfied. The reviewer is responsible to verify each aspect of the task.

- [ ] Any added or touched code follows our style-guide.

### CODE REVIEW

File: R/input.R [hunk 1]
```diff
@@ -212,7 +219,7 @@ numeric_input <- function(input_id, label, value, min = NA, max = NA, step = NA,
 #' @param ... Other parameters passed to \code{\link{numeric_input}} like \code{type} or \code{icon}.
 #' @rdname numeric_input
 #' @export
-numericInput <- function(inputId, label, value,
+numericInput <- function(inputId, label, value = NULL,
```
Comment (jakubnowicki) [inline R/input.R:222]: I would suggest adding `placeholder` as a named argument and adding the information about the requirement (needs value or placeholder) to the documentation.

File: R/input.R [hunk 2]
```diff
@@ -179,11 +179,18 @@ textInput <- function(inputId, label, value = "", width = NULL,
...
 #'
 #' @export
-numeric_input <- function(input_id, label, value, min = NA, max = NA, step = NA,
+numeric_input <- function(input_id, label, value = NULL, min = NA, max = NA, step = NA,
                            type = NULL, icon = NULL, placeholder = NULL, ...) {
-  if (!is.numeric(value) & !grepl("^\\d*(\\.\\d*|)$", value)) stop("Non-numeric input detected")
+  if (!(is.null(placeholder) || is.character(placeholder))) {
+    stop ("'placeholder' should be NULL or character")
+  }
+  if (is.null(value) & is.null(placeholder)) stop ("either 'value' or 'placeholder' should be defined")
```
Comment (jakubnowicki) [inline R/input.R:187]: This requirement should be also included in the function documentation.

File: R/input.R (hunk 1 above)
Comment (osenan) [reply R/input.R]: Done

File: R/input.R (hunk 2 above)
Comment (osenan) [reply R/input.R:188]: Done

**Review by jakubnowicki — APPROVED** [review]
LGTM

**Pull request merged**